opentaws/src/aircraft_state.rs

```diff
@@ -0,0 +1,525 @@
+
+    // Setterss
+    pub fn timestamp_mut(&mut self) -> &mut Time {
+        &mut self.timestamp
+    }
+
+    pub fn position_latitude_mut(&mut self) -> &mut Angle {
+        &mut self.position_lat
+    }
+
+    pub fn position_longitude_mut(&mut self) -> &mut Angle {
+        &mut self.position_lon
+    }
+
+    pub fn altitude_mut(&mut self) -> &mut Length {
+        &mut self.altitude
+    }
+
+    pub fn altitude_ground_mut(&mut self) -> &mut Length {
+        &mut self.altitude_ground
+    }
+
+    pub fn speed_ground_mut(&mut self) -> &mut Velocity {
+        &mut self.speed_ground
+    }
+
+    pub fn speed_air_mut(&mut self) -> &mut Velocity {
+        &mut self.speed_air
+    }
+
+    pub fn climb_rate_mut(&mut self) -> &mut Velocity {
+        &mut self.climb_rate
+    }
+
+    pub fn heading_mut(&mut self) -> &mut Angle {
+        &mut self.heading
+    }
+
+    pub fn track_mut(&mut self) -> &mut Angle {
+        &mut self.track
+    }
+
+    /* pub fn pitch_mut(&mut self) -> &mut Angle {
+        &mut self.pitch
+    }
+
+    pub fn roll_mut(&mut self) -> &mut Angle {
+        &mut self.roll
+    }
+
+    pub fn yaw_mut(&mut self) -> &mut Angle {
+        &mut self.yaw
+    } */
+
+    pub fn situation_mut(&mut self) -> &mut Option<FlightSegment> {
+        &mut self.situation
+    }
+}
+
+impl<T> AircraftState<T> {
+    // Getters
+    pub fn timestamp(&self) -> Time {
+        self.timestamp
+    }
+
+    pub fn position_latitude(&self) -> Angle {
+        self.position_lat
+    }
+
+    pub fn position_longitude(&self) -> Angle {
+        self.position_lon
+    }
+
+    pub fn altitude(&self) -> Length {
+        self.altitude
+    }
+
+    pub fn altitude_ground(&self) -> Length {
+        self.altitude_ground
+    }
+
+    pub fn speed_ground(&self) -> Velocity {
+        self.speed_ground
+    }
+
+    pub fn speed_air(&self) -> Velocity {
+        self.speed_air
+    }
+
+    pub fn climb_rate(&self) -> Velocity {
+        self.climb_rate
+    }
+
+    pub fn heading(&self) -> Angle {
+        self.heading
+    }
+
+    pub fn track(&self) -> Angle {
+        self.track
+    }
+
+    /* pub fn pitch(&self) -> Angle {
+        self.pitch
+    }
+
+    pub fn roll(&self) -> Angle {
+        self.roll
+    }
+
+    pub fn yaw(&self) -> Angle {
+        self.yaw
+    } */
+
+    pub fn situation(&self) -> &Option<FlightSegment> {
+        &self.situation
+    }
+
+    /// Normalizes the [AircraftState] into a [NormalizedAircraftState]
+    pub fn normalize(&self) -> NormalizedAircraftState {
+        let (lat, lon) = Self::wrap_position(self.position_lat, self.position_lon);
+
+        NormalizedAircraftState {
+            timestamp: self.timestamp,
+
+            position_lat: lat,
+            position_lon: lon,
+
+            altitude: self.altitude,
+            altitude_ground: Self::clamp_altitude_ground(self.altitude_ground),
+
+            speed_ground: self.speed_ground,
+            speed_air: self.speed_air,
+
```

Comment:
Would it make sense to map the values here to positive speed only? The idea is, we already know the direction in which the aircraft moves, so having the speed really should be an absolute.

Reply:
Yes, I think we should change the AircraftState in the next PR to contain a Position, Velocity and Acceleration vector and get rid of these scalar values.